dashboard/test/models/user_test.rb

```diff
@@ -2186,4 +2186,225 @@ class SectionCourses < ActiveSupport::TestCase
+      assert_equal [@script.id, @script2.id], student.get_hidden_script_ids(@course)
+    end
+
+    test "user in no sections" do
+      student = create :student
+
+      assert_equal [], student.get_hidden_stage_ids(@script.name)
+    end
+
+    test "teacher gets hidden stages for sections they own" do
+      teacher = create :teacher
+      teacher_teacher = create :teacher
+      student = create :student
+
+      teacher_owner_section = put_student_in_section(student, teacher, @script)
+      teacher_owner_section2 = put_student_in_section(student, teacher, @script)
+      teacher_member_section = put_student_in_section(teacher, teacher_teacher, @script)
+
+      # stage 1 is hidden in the first section owned by the teacher
+      SectionHiddenStage.create(section_id: teacher_owner_section.id, stage_id: @stage1.id)
+
+      # stage 1 and 2 are hidden in the second section owned by the teacher
+      SectionHiddenStage.create(section_id: teacher_owner_section2.id, stage_id: @stage1.id)
+      SectionHiddenStage.create(section_id: teacher_owner_section2.id, stage_id: @stage2.id)
+
+      # stage 3 is hidden in the section in which the teacher is a member
+      SectionHiddenStage.create(section_id: teacher_member_section.id, stage_id: @stage3.id)
+
+      # only the stages hidden in the owned section are considered hidden
+      expected = {
+        teacher_owner_section.id => [@stage1.id],
+        teacher_owner_section2.id => [@stage1.id, @stage2.id]
+      }
+      assert_equal expected, teacher.get_hidden_stage_ids(@script.id)
+    end
+
+    test "teacher gets hidden scripts for sections they own" do
+      teacher = create :teacher
+      teacher_teacher = create :teacher
+      student = create :student
+
+      teacher_owner_section = put_student_in_section(student, teacher, @script, @course)
+      teacher_owner_section2 = put_student_in_section(student, teacher, @script, @course)
+      teacher_member_section = put_student_in_section(teacher, teacher_teacher, @script, @course)
+
+      # stage 1 is hidden in the first section owned by the teacher
+      SectionHiddenScript.create(section_id: teacher_owner_section.id, script_id: @script.id)
+
+      # stage 1 and 2 are hidden in the second section owned by the teacher
+      SectionHiddenScript.create(section_id: teacher_owner_section2.id, script_id: @script.id)
+      SectionHiddenScript.create(section_id: teacher_owner_section2.id, script_id: @script2.id)
+
+      # stage 3 is hidden in the section in which the teacher is a member
+      SectionHiddenScript.create(section_id: teacher_member_section.id, script_id: @script3.id)
+
+      # only the scripts hidden in the owned section are considered hidden
+      expected = {
+        teacher_owner_section.id => [@script.id],
+        teacher_owner_section2.id => [@script.id, @script2.id]
+      }
+      assert_equal expected, teacher.get_hidden_script_ids(@course)
+    end
+  end
```

Comment:
This are mostly tests that had existed elsewhere, but cleaned up a little bit to repeat less. I then duplicated them to have the same logic we used to check for hidden stages in a script to instead check for hidden scripts in a course.